src/smile_server.py

```diff
@@ -13,35 +13,57 @@
 # This server parses base64 encoded png images and runs them
 # through the preprocessor and the verifier and should eventually
 # be able to recognize identities
-class SmileHttpHandler(BaseHTTPRequestHandler):
+class VisionHandler(BaseHTTPRequestHandler):
     preprocessor = None
-    verifier = None
+    model = None
 
     def img_from_b64(self, b64_str):
         b_img = base64.b64decode(b64_str)
         nparr = np.fromstring(b_img, np.uint8)
         return cv2.imdecode(nparr, cv2.IMREAD_COLOR)
 
-    def verify_img(self):
+    def get_fields(self):
         form_data = self.rfile.read(int(self.headers["Content-Length"]))
-        fields = urlparse.parse_qs(form_data)
+        return urlparse.parse_qs(form_data)
+
+    def read_img(self, fields):
         b64_str= fields['image'][0]
-        img = self.img_from_b64(b64_str)
+        return self.img_from_b64(b64_str)
+
+    def verify_img(self):
+        fields = self.get_fields()
+        img = self.read_img(fields)
         processed = self.preprocessor.process(img)
         if len(processed) > 1:
             print("multiple faces")
-        identity = self.verifier.verify(processed[0])
-        # return identity to user
-        return
+        identity = self.model.verify(processed[0])
+        return identity
+
+    def train(self):
+        fields = self.get_fields()
+        img = self.read_img(fields)
+        model.train(img, fields['label'])
+        return 200
 
```

Comment:
@hkdsun @seeARMS confirm how requests to vision look and how responses to app should be structured
suggesting:
Requests `POST /verify?img=b64img` and `POST /train?img=b64img&label=string`
Responses `200 name` and `200` where name = [valid name] | `no_match`